docs/antrea-ipam.md

```diff
@@ -46,3 +46,131 @@ cluster. Valid range is 16 to 30. Default is 24.
+The pool can be annotated to Namespace, and Antrea will manage IP address assignment for
+corresponding Pods according to pool spec. In the future, support will be extended to
+Deployments and StatefulSets, as well as requesting specific IP via Pod annotation.
+
+Note that IP pool annotation can not be updated, but rather re-created. IP Pool can be
+extended, but cannot be shrunk if already assigned to a resource. The IP ranges of IP
+Pools must not overlap, otherwise it would lead to undefined behavior.
+
+Regular `Subnet per Node` IPAM will continue to be used for resources without IP pool
+annotation, or when `AntreaIPAM` feature is disabled.
+
+### Configuration
+
+Users need to enable `AntreaIPAM` from the featureGates map defined in antrea.yml for
+both Controller and Agent:
+
+```yaml
+  antrea-controller.conf: |
+    featureGates:
+    # Enable flexible IPAM mode for Antrea. This mode allows to assign IP Ranges to Namespaces,
+    # Deployments and StatefulSets via IP Pool annotation.
+      AntreaIPAM: true
+  antrea-agent.conf: |
+    featureGates:
+    # Enable flexible IPAM mode for Antrea. This mode allows to assign IP Ranges to Namespaces,
+    # Deployments and StatefulSets via IP Pool annotation.
+      AntreaIPAM: true
+```
+
+Usage example:
+
+```yaml
+apiVersion: "crd.antrea.io/v1alpha2"
+kind: IPPool
+metadata:
+  name: pool1
+spec:
+  ipVersion: 4
+  ipRanges:
+  - start: "10.2.0.12"
+    end: "10.2.0.20"
+    gateway: "10.2.0.1"
+    prefixLength: 24
+```
+
+```yaml
+kind: Namespace
+metadata:
+  annotations:
+    ipam.antrea.io/ippools: 'pool1'
+```
+
+### Data path change for this feature
+
+When `AntreaIPAM` is enabled, `antrea-agent` will connect the Node's network interface
+to the OVS bridge at startup, and it will detach the interface from the OVS bridge and
+restore its configurations at exit. Node may lose network connection when `antrea-agent`
+or OVS daemons are stopped unexpectedly, which can be recovered by rebooting the Node.
+`AntreaIPAM` Pods' traffic will not be routed by local Node's network stack.
+
+All traffic to a local Pod will be sent to the Pod's OVS port directly, after the
+destination MAC is rewritten to the Pod's MAC address. This includes `AntreaIPAM` Pods
+and regular `Subnet per Node` IPAM Pods, even they are not in the same subnets.
+Inter-Node traffic will be sent to the Node network from the source Node, and forwarded
+to the destination Node by the Node network.
+
+### Requirements for this Feature
+
+In Antrea 1.4, this feature is supported on Linux Nodes, with IPv4, `system` OVS datapath
+type, and `noEncap`, `noSNAT` traffic mode.
+
+The IPs in the `IPPools` must be in the same "underlay" subnet as the Node IP, because
+inter-Node traffic of AntreaIPAM Pods is forwarded by the Node network. Only a single IP
+pool can be included in the Namespace annotation. In the future, annotation of up to two
+pools for IPv4 and IPv6 respectively will be supported.
+
+### Updates in Antrea 1.5
```

Comment:
We should just talk about all features together, and indicate some features are available from which release; rather than creating a separate section for each release. We should make users easy to see all latest features.

Reply:
Re-structured this part.

Reply:
Maybe we can add a section about "Usage", and put all IPPool, Namespace, Pod, Deployment, and StatefulSet examples there.

Reply:
Fixed.